pkg/ddl/table.go

```diff
@@ -970,22 +969,13 @@ func finishJobRenameTables(jobCtx *jobContext, job *model.Job, args *model.Renam
 	// Before updating the schema version, we need to reset the old schema ID to new schema ID, so that
 	// the table info can be dropped normally in `ApplyDiff`. This is because renaming table requires two
 	// schema versions to complete.
-	// TODO(joechenrh): set the old schemaID in Args is a bit hacky. Maybe we need a better solution.
-	var err error
-	oldRawArgs := job.RawArgs
 	for _, info := range infos {
-		info.OldSchemaID = info.NewSchemaID
-	}
-	job.FillArgs(args)
-	job.RawArgs, err = json.Marshal(job.Args)
-	if err != nil {
-		return 0, errors.Trace(err)
+		info.OldSchemaIDForSchemaDiff = info.NewSchemaID
```

Comment:
after reading the comment I still don't understand why we should set `OldSchemaIDForSchemaDiff` instead of `OldSchemaID`. What does the unchanged `OldSchemaID` use for?

Reply:
see this https://github.com/pingcap/tidb/pull/44585/files#diff-9dd5a0976146b94bb3074ca04e8131dffc03096505755440e64760b42bbde0fdR1181-R1183

that fix is backup RawArgs, change args and put into RawArgs, so later decode can see it, then restore

here we use OldSchemaIDForSchemaDiff to do the same thing by pass is directly

Reply:
the fix in #44585 should better be done in diff apply, not here, but I won't change it here

Reply:
Maybe remove the comments above since job args are not changed?

Reply:
those comments explains why we do it, so I keep it